src/navigation/MainTabNavigator.tsx

```diff
@@ -1,37 +1,154 @@
 import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 
-import { colors, device, normalize } from '../config';
-import { TabNavigatorConfig } from '../types';
+import { LoadingSpinner } from '../components';
+import { colors, normalize } from '../config';
+import {
+  createDynamicTabConfig,
+  tabNavigatorConfig as defaultTabNavigatorConfig
+} from '../config/navigation/tabConfig';
+import { useStaticContent } from '../hooks';
+import { ScreenName, TabConfig, TabNavigatorConfig } from '../types';
 
 import { getStackNavigator } from './AppStackNavigator';
 
+const defaultTabRoutes: TabRoutes = {
+  tabConfig: {
+    activeBackgroundColor: colors.surface,
+    activeTintColor: colors.accent,
+    inactiveBackgroundColor: colors.surface,
+    inactiveTintColor: colors.primary,
+    tabs: ['Home', 'Service', 'Events', 'Index', 'About']
+  }
+};
+
+type Tab = {
+  accessibilityLabel: string;
+  iconName: string;
+  iconSize: number;
+  label: string;
+  screen: ScreenName;
+  params?: Record<string, any>;
+};
+
+type TabRoutes = {
+  tabConfig: {
+    activeTintColor: string;
+    inactiveTintColor: string;
+    activeBackgroundColor: string;
+    inactiveBackgroundColor: string;
+    tabs: (Tab | string)[];
+  };
+};
+
+const useTabRoutes = () => {
+  const { data, error, loading } = useStaticContent<TabRoutes>({
+    name: 'tabNavigation',
+    type: 'json'
+  });
+
+  const [tabRoutes, setTabRoutes] = useState<TabRoutes>(defaultTabRoutes);
+
+  useEffect(() => {
+    const tabs: (Tab | string)[] = data?.tabConfig.tabs || defaultTabRoutes.tabConfig.tabs;
+    const { activeBackgroundColor, activeTintColor, inactiveBackgroundColor, inactiveTintColor } =
+      data?.tabConfig || defaultTabRoutes.tabConfig;
+
+    setTabRoutes((prev) => {
+      const filteredTabs = defaultTabNavigatorConfig.tabConfigs.filter((tabConfig) => {
+        const screen = tabConfig.stackConfig.initialRouteName;
+
+        return tabs.some((tab) =>
+          typeof tab === 'string' ? tab === screen : tab.screen === screen
+        );
+      });
```

Comment:
what is happening here in human words? filtering tabs with a `some` inside for what exact purpose?

Reply:
Actually I don't think we need this `filteredTabs` anymore.

To explain the code here:

default scrolls through the tabs and filters by `initialRouteName`. Then it uses this filtering in `dynamicTabs`. But it also works if we apply filtering directly in `dynamicTabs`. This is just creating confusion.